lib/AST/ASTMangler.cpp

```diff
@@ -2999,12 +3017,53 @@ void ASTMangler::appendTypeListElement(Identifier name, Type elementType,
   if (flags.isCompileTimeConst())
     appendOperator("Yt");
 
+  if (lifetimeDependenceKind) {
+    appendLifetimeDependenceKind(*lifetimeDependenceKind,
+                                 /*isSelfDependence*/ false);
+  }
+
   if (!name.empty())
     appendIdentifier(name.str());
   if (flags.isVariadic())
     appendOperator("d");
 }
 
+void ASTMangler::appendLifetimeDependenceKind(LifetimeDependenceKind kind,
+                                              bool isSelfDependence) {
+  // If we converge on dependsOn(borrowed: paramName)/dependsOn(borrowed:
+  // paramName) syntax, this can be a single case value check.
+  if (kind == LifetimeDependenceKind::Borrow ||
+      kind == LifetimeDependenceKind::Mutate) {
+    if (isSelfDependence) {
+      appendOperator("YLs");
+    } else {
+      appendOperator("Yls");
+    }
+  } else {
+    // If we converge on dependsOn(borrowed: paramName)/dependsOn(borrowed:
```

Comment:
I think you meant `dependsOn(borrowed: paramName)/dependsOn(paramName`

Reply:
Yes! If no other comments, I'll do this as a follow on.